hostfactory/host_provider/src/allocation_strategy.py

```diff
@@ -0,0 +1,228 @@
+from math import ceil, floor
+from enum import Enum
+from hpc.autoscale.util import partition, partition_single
+import logging
+
+
+class AllocationStrategies(Enum):
+    PRICE = "price"
+    CAPACITY = "capacity"
+    DECAY = "decay"
+    WEIGHTED = "weighted"
+
+
+def round_down_to_nearest_multiple(target, x):
+    return x * floor(target / x)
+
+def round_up_to_nearest_multiple(target, x):
+    return x * ceil(target / x)
+
+def calculate_vm_dist_capacity(vm_types, total_slot_count):
+    vm_dist = {}
+    slot_counts = [total_slot_count // len(vm_types) for n in range(len(vm_types))]
```

Comment:
Why for n in range(len(vm_types)) if you aren't actually using n for anything?
This is equivalent
[total_slout_count // len(vm_types)] * len(vm_types)
But then, why not just have slot_count = total_slot_count // len(vm_types)?